src/utils/transforms.cljs

```diff
@@ -21,3 +24,37 @@
   [json]
   (when-not (= json "undefined")
     (try (.parse js/JSON json) (catch js/Error _ (when (string? json) json)))))
+
+;; Functions took from reagent and modified to also transform vectors contained in styles
+
+(def ^:private js-val? reagent.impl.util/js-val?)
+(def ^:private named? reagent.impl.util/named?)
+(def ^:private cached-prop-name reagent.impl.template/cached-prop-name)
```

Comment:
The code looked so ugly having those long names, that's why I added it. but I can just use them inline. wdyt?

Reply:
I imagined you did that for aesthetics o/ I find the code without the unnecessary indirection clearer, especially because they are used only once. For instance, when I saw `js-val?` and `named?` I didn't know what these functions were or if they were from cljs.core, but to my surprise they were just renames, this took me some time to figure out, others reading the code would have to do the same. You could try to shorten the namespace a little bit, like `reagent.impl.util` -> `reagent.util` (that `impl` doesn't add anything for readability).

With the aliased namespace it reads just as beautifully to me, but less code.

```clojure
(defn styles-with-vectors
  [x]
  (cond (reagent.util/js-val? x) x
        (reagent.util/named? x)  (name x)
        (map? x)                 (reduce-kv kv-conv #js {} x)
        (vector? x)              (to-array (map styles-with-vectors x))
        (coll? x)                (clj->js x)
        (ifn? x)                 (fn [& args]
                                   (apply x args))
        :else                    (clj->js x)))
```

Reply:
Applied @ilmotta !

Also added some docstrings as @OmarBasem suggested.